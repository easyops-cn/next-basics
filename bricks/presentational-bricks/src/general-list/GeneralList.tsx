import React from "react";
import classNames from "classnames";

interface GeneralListProps {
  isCardList: boolean;
  cardWidth?: string;
  cardMinWidth?: string;
}

export function GeneralList(props: GeneralListProps): React.ReactElement {
  return (
    <div
      className={classNames("listContainer", {
        cardListContainer: props.isCardList
      })}
      style={
        props.cardMinWidth
          ? {
              gridTemplateColumns: `repeat(auto-fill,minmax(${props.cardMinWidth}, 1fr))`
            }
          : props.cardWidth
          ? {
              gridTemplateColumns: `repeat(auto-fill,${props.cardWidth})`
            }
          : {}
      }
    >
      <slot id="itemsSlot" name="items" />
    </div>
  );
}
