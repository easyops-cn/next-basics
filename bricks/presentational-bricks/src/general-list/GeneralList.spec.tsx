import React from "react";
import { shallow } from "enzyme";
import { GeneralList } from "./GeneralList";

describe("GeneralList", () => {
  it("should work", () => {
    const wrapper = shallow(<GeneralList isCardList={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
