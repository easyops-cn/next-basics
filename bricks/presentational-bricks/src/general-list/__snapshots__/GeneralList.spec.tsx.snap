// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralList should work 1`] = `
<div
  className="listContainer cardListContainer"
  style={Object {}}
>
  <slot
    id="itemsSlot"
    name="items"
  />
</div>
`;
